lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -656,6 +659,57 @@ sub clear_context {
     $adapter_context= undef;
 }
 
+=head2 mask_sensitive
+
+Mask sensitive data in the message and logs error in case of failure
+
+=over 4
+
+=item * C<$message> string - The message to be masked
+
+=back
+
+Returns string - The masked message
+
+=cut
+
+sub mask_sensitive {
+    my ($message) = @_;
+
+    # Define a lookup hash reference for all sensitive data regex patterns to be logged
+    my $sensitive_patterns = {
+        'PII' => [
+            qr/\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b/,  # Email
+        ],
+        'Sensitive' => [
+            qr/\b(?:token|key|oauth[ _-]?token)\s*[:=]\s*([^\s]+)/i, # token or api key
+        ],
+        'Token' => [
+            qr/a1\-[a-zA-Z0-9\-]{29}/,  #oauth Token pattern
+            qr/[a-zA-Z0-9\-]{15}/, #API Token pattern
```

Comment:
If I'm not mistaken, there's no `-` in `OAuth token` and same for `API token` [Link](https://github.com/regentmarkets/bom-platform/blob/648910721261a212b4aad90d03b982ffab6f522d/lib/BOM/Platform/Token/API.pm#L307). Let's consider `Refresh and CTrader tokens` as well?
```suggestion
            qr/a1|r1|ct1\-[a-zA-Z0-9]{29}/,  # OAuth, Refresh and CTrader token patterns
            qr/[a-zA-Z0-9]{15}/, #API Token pattern
```